provider_config.go

```diff
@@ -0,0 +1,57 @@
+package jwtauth
+
+import (
+	"fmt"
+)
+
+// Provider-specific configuration interfaces
+// All providers must implement the CustomProvider interface, and may implement
+// others as needed.
+
+// ProviderMap is used to map a provider name to its provider type
+var ProviderMap = map[string]CustomProvider{
+	// TODO: remove "empty" provider when actual providers are added
+	"empty": &EmptyProvider{},
+}
+
+// CustomProvider - Any custom provider must implement this interface
+type CustomProvider interface {
+	Initialize(*jwtConfig) error
+	SensitiveKeys() []string
```

Comment:
Can we document the specific responsibilities of each of these methods? We should establish what is expected of these methods so developers know what to do in order to build "well behaved" custom providers

Reply:
Added doc comments in fe083e8

Reply:
Excellent 👍